### DISCUSSION TITLE
Change source code uploaded by GitHub

### DISCUSSION THREAD

**Discussion by Xwilarg** [community/community#22712] [New to GitHub]
<p>Hi,</p>
<p>I would like to do a release on one of my project but by default GitHub add the source code of the current master branch.<br>
I would have like to know if I can change the commit targeted (or if there is just a way to not upload it)</p>
<p>Thanks in advance,</p>

**Comment by tajmone** [comment, marked as answer]
<aside class="quote no-group" data-username="Xwilarg" data-post="1" data-topic="127183">
<div class="title">
<div class="quote-controls"></div>
 Xwilarg:</div>
<blockquote>
<p>by default GitHub add the source code of the current master branch.</p>
</blockquote>
</aside>
<p>From <a href="https://docs.github.com/en/github/administering-a-repository/managing-releases-in-a-repository" rel="nofollow noopener">GitHub Help » Managing releases in a repository</a>:</p>
<p><img src="https://user-images.githubusercontent.com/11064292/181072678-1ea38876-888d-4655-b27f-28d85d480676.png" alt="screenshot" data-base62-sha1="wcceCTnsRRf4R0vWa6i0mjmXNci" width="630" height="276"></p>
<p>you can see that you can choose other branches from the <kbd>Target:</kbd> dropdown.</p>
<p>Also, releases are associated to <strong>tags</strong>, which are not limited to a specific branch. When you create a tag (whether from GitHub’s WebUI or via Git on the command line) GitHub will create a release matching the commit referenced by that tag.</p>
<aside class="quote no-group" data-username="Xwilarg" data-post="1" data-topic="127183">
<div class="title">
<div class="quote-controls"></div>
 Xwilarg:</div>
<blockquote>
<p>I would have like to know if I can change the commit targeted (or if there is just a way to not upload it)</p>
</blockquote>
</aside>
<p>You can always edit (or even delete) releases on your repository — although it’s preferable not to, unless strictly necessary, especially if some end users have already downloaded the release archive (i.e. it might leave contradicting release versions behind).</p>
<p>Creating a release isn’t really related to “uploading”, it’s about tagging a commit which is already in the repository (on GitHub). GitHub releases allow you to attach custom archives (Zip file) to any tagged release (e.g. pre-compiled binaries), but these are “extra”, and the core of the release is the commit pointed to by tagged reference.</p>
